ChildProject/annotations.py

```diff
@@ -0,0 +1,287 @@
+        self.annotations = table.read()
+        errors, warnings = table.validate()
+        return errors, warnings
+
+    def validate(self):
+        errors, warnings = [], []
+
+        for annotation in self.annotations.to_dict(orient = 'records'):
+            segments = IndexTable(
+                'segments',
+                path = os.path.join(self.project.path, 'annotations', annotation['annotation_filename']),
+                columns = self.SEGMENTS_COLUMNS
+            )
+
+            segments.read()
+            res = segments.validate()
+            errors += res[0]
+            warnings += res[1]
+
+        return errors, warnings
+        
+
+    def load_textgrid(self, filename):
+        path = os.path.join(self.project.path, 'raw_annotations', filename)
+        textgrid = pympi.Praat.TextGrid(path)
+
+        segments = []
+        for tier in textgrid.tiers:
+            for interval in tier.intervals:
+                tier_name = tier.name.strip()
+
+                if tier_name == 'Autre':
+                    continue
+
+                if interval[2] == "":
+                    continue
+
+                segment = {
+                    'segment_onset': "{:.3f}".format(interval[0]),
+                    'segment_offset': "{:.3f}".format(interval[1]),
+                    'speaker_id': tier_name,
+                    'ling_type': interval[2] if interval[2] else "",
+                    'speaker_type': self.SPEAKER_ID_TO_TYPE[tier_name] if tier_name in self.SPEAKER_ID_TO_TYPE else 'NA',
+                    'vcm_type': 'NA',
+                    'lex_type': 'NA',
+                    'mwu_type': 'NA',
+                    'addresseee': 'NA',
+                    'transcription': 'NA'
+                }
+
+                segments.append(segment)
+
+        return pd.DataFrame(segments)
+
+    def load_eaf(self, filename):
+        path = os.path.join(self.project.path, 'raw_annotations', filename)
+        eaf = pympi.Elan.Eaf(path)
+
+        segments = {}
+
+        for tier_name in eaf.tiers:
+            annotations = eaf.tiers[tier_name][0]
+
+            for aid in annotations:
+                (start_ts, end_ts, value, svg_ref) = annotations[aid]
+                (start_t, end_t) = (eaf.timeslots[start_ts], eaf.timeslots[end_ts])
+
+                segment = {
+                    'segment_onset': "{:.3f}".format(start_t/1000),
+                    'segment_offset': "{:.3f}".format(end_t/1000),
+                    'speaker_id': tier_name,
+                    'ling_type': 'NA',
+                    'speaker_type': self.SPEAKER_ID_TO_TYPE[tier_name] if tier_name in self.SPEAKER_ID_TO_TYPE else 'NA',
+                    'vcm_type': 'NA',
+                    'lex_type': 'NA',
+                    'mwu_type': 'NA',
+                    'addresseee': 'NA',
+                    'transcription': value if value != '0.' else '0'
```

Comment:
Should we really translate 0. to 0 ? This was necessary for the tests to pass

Reply:
this is a mistake in the google spreadsheet, which autocorrected 0 to 0., when in fact,  the correct entry for transcription is 0. (with the stop). So if anything, we should do the opposite: it's a common error for humans to forget to write the stop.